pkgs/development/idris-modules/default.nix

```diff
@@ -25,22 +25,35 @@
       pruviloj = [ self.prelude self.base ];
     };
 
-    files = builtins.filter (n: n != "default") (pkgs.lib.mapAttrsToList (name: type: let
-      m = builtins.match "(.*)\\.nix" name;
-    in if m == null then "default" else builtins.head m) (builtins.readDir ./.));
-  in (builtins.listToAttrs (map (name: {
-    inherit name;
-
-    value = callPackage (./. + "/${name}.nix") {};
-  }) files)) // {
```

Comment:
We can certainly clean this up/document it more, but :-1: for me on removing the autodiscovery.

Reply:
I don't like this style at all, it is too implicit and relied on a false assumption that you want an attribute for every `*.nix` file in a repo. There are not many places in nixpkgs which I could find using it.